AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserViewModel.kt

```diff
@@ -604,6 +608,41 @@ class CardBrowserViewModel(
         }
     }
 
+    /**
+     * Obtains data to be displayed to the user then sent to [repositionSelectedRows]
+     */
+    suspend fun prepareToRepositionCards(): RepositionCardsRequest {
+        val selectedCardIds = queryAllSelectedCardIds()
+        // Only new cards may be repositioned (If any non-new found show error dialog and return false)
+        if (selectedCardIds.any { withCol { getCard(it).queue != QueueType.New } }) {
+            return RepositionCardsRequest.ContainsNonNewCardsError
+        }
+
+        // query obtained from Anki Desktop
+        // https://github.com/ankitects/anki/blob/1fb1cbbf85c48a54c05cb4442b1b424a529cac60/qt/aqt/operations/scheduling.py#L117
+        try {
+            val (min, max) =
+                withCol {
+                    db.query("select min(due), max(due) from cards where type=${CardType.New.code} and odid=0").use {
```

Comment:
Untested, but I believe that is preferable to not add the values to a SQL string manually.

```suggestion
                    db.query("select min(due), max(due) from cards where type=? and odid=0", CardType.New.code).use {
```